app/src/main/java/org/dhis2/usescases/searchTrackEntity/listView/SearchTEList.kt

```diff
@@ -0,0 +1,206 @@
+package org.dhis2.usescases.searchTrackEntity.listView
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.activityViewModels
+import androidx.recyclerview.widget.ConcatAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.composethemeadapter.MdcTheme
+import javax.inject.Inject
+import org.dhis2.commons.animations.hideWithTranslation
+import org.dhis2.commons.animations.showWithTranslation
+import org.dhis2.commons.bindings.clipWithRoundedCorners
+import org.dhis2.databinding.FragmentSearchListBinding
+import org.dhis2.usescases.general.FragmentGlobalAbstract
+import org.dhis2.usescases.searchTrackEntity.CreateNewButton
+import org.dhis2.usescases.searchTrackEntity.SearchButton
+import org.dhis2.usescases.searchTrackEntity.SearchList
+import org.dhis2.usescases.searchTrackEntity.SearchTEActivity
+import org.dhis2.usescases.searchTrackEntity.SearchTEContractsModule
+import org.dhis2.usescases.searchTrackEntity.SearchTEIViewModel
+import org.dhis2.usescases.searchTrackEntity.SearchTeiViewModelFactory
+import org.dhis2.usescases.searchTrackEntity.adapters.SearchTeiLiveAdapter
+
+const val ARG_FROM_RELATIONSHIP = "ARG_FROM_RELATIONSHIP"
+
+class SearchTEList : FragmentGlobalAbstract() {
+
+    @Inject
+    lateinit var viewModelFactory: SearchTeiViewModelFactory
+
+    @Inject
+    lateinit var presenter: SearchTEContractsModule.Presenter
```

Comment:
Could this presenter be extracted to the viewModel?

Reply:
> 

Done!